types/matches.d.ts

```diff
@@ -1,6 +1,14 @@
+import { ARIARole } from 'aria-query'
+
 export type MatcherFunction = (content: string, element: HTMLElement) => boolean
 export type Matcher = string | RegExp | MatcherFunction
 
+// important to use String here
+// this gives us intellisense but it can also accept values
+// that are not included in the union types
+// tslint:disable-next-line:ban-types
+export type ByRoleMatcher = ARIARole | String | RegExp | MatcherFunction
```

Comment:
Does this work with `string` as well?

```suggestion
export type ByRoleMatcher = ARIARole | string | RegExp | MatcherFunction
```

The comment explains that `String` is important but it's not obvious from the comment why `string` wouldn't work. This would be the more "natural" choice and get rid of the tslint warning.

Reply:
`string` doesn't give us the intellisense we want:

As `String`:

![image](https://user-images.githubusercontent.com/28659384/86011127-9b905700-ba1c-11ea-9614-ae528ac24e8b.png)

As `string`:

![image](https://user-images.githubusercontent.com/28659384/86011163-a5b25580-ba1c-11ea-8258-d78dcc175f37.png)

Reply:
Your comment made me thinking, and I changed the type implementation in the last commit.
WDYT?

It has the same result

![image](https://user-images.githubusercontent.com/28659384/86040212-eb364900-ba43-11ea-9db5-9b4bb04d74fa.png)

Reply:
Seems like the linter doesn't like this ...